torch_harmonics/s2_convolutions.py

```diff
@@ -95,21 +95,28 @@ def _precompute_convolution_tensor(
     out_vals = torch.empty([0], dtype=torch.long)
 
     # compute the phi differences
-    phis = torch.linspace(0, 2 * math.pi, nlon_in)
+    lons_in = torch.linspace(0, 2*math.pi, nlon_in+1)[:-1]
 
     for t in range(nlat_out):
-        alpha = -lats_in.reshape(-1, 1)
-        beta = phis
-        gamma = lats_out[t]
+        alpha = - lats_out[t]
+        beta = lons_in
+        gamma = lats_in.reshape(-1, 1)
 
         # compute latitude of the rotated position
-        z = torch.cos(alpha) * torch.cos(gamma) - torch.cos(beta) * torch.sin(alpha) * torch.sin(gamma)
-        z = torch.clamp(z, min=-1.0, max=1.0)
-        theta = torch.arccos(z)
+        z = - torch.cos(beta) * torch.sin(alpha) * torch.sin(gamma) + torch.cos(alpha) * torch.cos(gamma)
```

Comment:
can we write a short derivation of this stuff as a comment or dosctring or so. it also helps us to remember what we did a few months later :)